charts/m4d/templates/webhook-configs.yaml

```diff
@@ -1,89 +1,3 @@
-apiVersion: admissionregistration.k8s.io/v1beta1
-kind: MutatingWebhookConfiguration
-metadata:
-  creationTimestamp: null
-  name: '{{ .Release.Namespace }}-mutating-webhook'
-  annotations:
-    cert-manager.io/inject-ca-from: '{{ .Release.Namespace }}/serving-cert'
-webhooks:
-  - clientConfig:
-      caBundle: Cg==
-      service:
-        name: webhook-service
-        namespace: '{{ .Release.Namespace }}'
-        path: /mutate-motion-m4d-ibm-com-v1alpha1-batchtransfer
-    failurePolicy: Fail
-    name: mbatchtransfer.kb.io
-    rules:
-      - apiGroups:
-          - motion.m4d.ibm.com
-        apiVersions:
-          - v1alpha1
-        operations:
-          - CREATE
-          - UPDATE
-        resources:
-          - batchtransfers
-  - clientConfig:
-      caBundle: Cg==
-      service:
-        name: webhook-service
-        namespace: '{{ .Release.Namespace }}'
-        path: /mutate-motion-m4d-ibm-com-v1alpha1-streamtransfer
-    failurePolicy: Fail
-    name: mstreamtransfer.kb.io
-    rules:
-      - apiGroups:
-          - motion.m4d.ibm.com
-        apiVersions:
-          - v1alpha1
-        operations:
-          - CREATE
-          - UPDATE
-        resources:
-          - streamtransfers
----
-apiVersion: admissionregistration.k8s.io/v1beta1
-kind: ValidatingWebhookConfiguration
-metadata:
-  creationTimestamp: null
-  name: '{{ .Release.Namespace }}-validating-webhook'
-  annotations:
-    cert-manager.io/inject-ca-from: '{{ .Release.Namespace }}/serving-cert'
-webhooks:
-  - clientConfig:
-      caBundle: Cg==
-      service:
-        name: webhook-service
-        namespace: '{{ .Release.Namespace }}'
-        path: /validate-motion-m4d-ibm-com-v1alpha1-batchtransfer
-    failurePolicy: Fail
-    name: vbatchtransfer.kb.io
-    rules:
-      - apiGroups:
-          - motion.m4d.ibm.com
-        apiVersions:
-          - v1alpha1
-        operations:
-          - CREATE
-          - UPDATE
-        resources:
-          - batchtransfers
-  - clientConfig:
-      caBundle: Cg==
-      service:
-        name: webhook-service
-        namespace: '{{ .Release.Namespace }}'
-        path: /validate-motion-m4d-ibm-com-v1alpha1-streamtransfer
-    failurePolicy: Fail
-    name: vstreamtransfer.kb.io
-    rules:
-      - apiGroups:
-          - motion.m4d.ibm.com
-        apiVersions:
-          - v1alpha1
-        operations:
-          - CREATE
-          - UPDATE
-        resources:
-          - streamtransfers
+{{- if include "m4d.isEnabled" (tuple .Values.manager.enabled (or .Values.coordinator.enabled .Values.worker.enabled)) }}
```

Comment:
Currently the webhooks are only used in the motion apis. Thus only in the worker. But if we're planning to add webhooks for the other APIs as well we should leave this as is. (And I think we can find some useful checks for M4dApplication as well)

Reply:
Yes I wondered whether we want these always enabled to avoid changes here in the future. What would be your preference?

Reply:
I would leave them enabled for all.